Tools/RemoteSynchronization/Program.cs

```diff
@@ -0,0 +1,815 @@
+                {
+                    sw_get_src.Start();
+                    await b_src.GetAsync(f.Name, s_src, CancellationToken.None);
+                    sw_get_src.Stop();
+                    if (GlobalConfig.DryRun)
+                    {
+                        Duplicati.Library.Logging.Log.WriteDryrunMessage(LOGTAG, "rsync", "Would write {0} bytes of {1} to {2}", s_src.Length, f.Name, b_dst.DisplayName);
+                    }
+                    else
+                    {
+                        sw_put_dst.Start();
+                        await b_dst.PutAsync(f.Name, s_src, CancellationToken.None);
+                        sw_put_dst.Stop();
+                        if (GlobalConfig.VerifyGetAfterPut)
+                        {
+                            sw_get_dst.Start();
+                            await b_dst.GetAsync(f.Name, s_dst, CancellationToken.None);
+                            sw_get_dst.Stop();
+
+                            sw_get_cmp.Start();
+                            string? err_string = null;
+                            if (s_src.Length != s_dst.Length)
+                            {
+                                err_string = $"The sizes of the files do not match: {s_src.Length} != {s_dst.Length}.";
+                            }
+
+                            if (!s_src.ToArray().SequenceEqual(s_dst.ToArray()))
+                            {
+                                err_string = (err_string is null ? "" : err_string + " ") + "The contents of the files do not match.";
+                            }
+                            sw_get_cmp.Stop();
+
+                            if (err_string is not null)
+                            {
+                                throw new Exception(err_string);
+                            }
+                        }
+                    }
+
+                    successful_copies++;
+                }
+                catch (Exception e)
+                {
+                    Duplicati.Library.Logging.Log.WriteErrorMessage(LOGTAG, "rsync", e, "Error copying {0}: {1}", f.Name, e.Message);
+                    errors.Add(f);
+                }
+                finally
+                {
+                    s_src.SetLength(0);
+                    s_dst.SetLength(0);
+                    i++;
+
+                    // Stop any running timers
+                    sw_get_src.Stop();
+                    sw_put_dst.Stop();
+                    sw_get_dst.Stop();
+                    sw_get_cmp.Stop();
+                }
+            }
+
+            if (GlobalConfig.Progress)
+                Console.WriteLine($"\rCopying: {n}/{n}");
+
+            Duplicati.Library.Logging.Log.WriteProfilingMessage(LOGTAG, "rsync", "Copy | Get source: {0} ms, Put destination: {1} ms, Get destination: {2} ms, Get compare: {3} ms", sw_get_src.ElapsedMilliseconds, sw_put_dst.ElapsedMilliseconds, sw_get_dst.ElapsedMilliseconds, sw_get_cmp.ElapsedMilliseconds);
```

Comment:
If you use `TimeSpan.FromMilliseconds(...)` the logged values will be easier to read than the raw millisecond values.